### ISSUE TITLE
a request for more detailed readme file 

### ISSUE THREAD

**Issue by sephirxth** [InputUsername/Whiteboard#4]
Dear developers, 

I would like to have a tool to help organize my usual piece of thinking, I tried a lot of software, but they can not meet my needs. So I’m going to try to develop such a tool on my own.

I want to start with this project of yours. I have a question and a request I’d like to hear from you.

1. [question] Could you explain why you went from js to rust to develop a whiteboard?

2. [request] I know a bit about programming, but I’m not a programmer. Could you please add more explanations on how to build this project and its structure?

Thanks a lot.

**Comment by InputUsername** [comment]
Hi! I'm not sure what exactly you would want to use this program for. I initially made it as a fun weekend project to dive into web development. It's a very simply app with a server that multiple people can connect to from a webpage, to draw on a virtual whiteboard together.

To answer your questions:
1. The JavaScript version of the app uses bad practices and was more a proof-of-concept than an actual good web app. It's very inefficient. Moreover, the server was made in an old version of Ruby, using an old unmaintained WebSocket library, and was also very inefficient. Rust is more modern, and allows compiling to WebAssembly. Wasm works similarly to JS, but is faster in some scenarios. Moreover the new server (also written in Rust) is much faster than the old Ruby server.
2. I'm afraid the new version is not in a working state, it allows connecting to the server and not much else. You can try the old version on the [master branch](https://github.com/InputUsername/whiteboard/tree/master), which has a decent Readme, but no guarantees that it works anymore. I'm not even sure which Ruby version I used at the time, maybe version 1.8.

So, in short: this probably isn't the right program for you (although feel free to fork it and make it work for your use case!). I may or may not work on it in the future but no guarantees.
